### PR TITLE
[Feat] Pagination for proposals and history

### PR DESCRIPTION
**Pull request by JulesGuesnon**
# Description

This PR adds pagination for proposals and history, and it fixes `0.3.1` related bug as well

### CODE REVIEW

File: versioned/version0_3_1.ts [hunk 1]
```diff
@@ -389,4 +389,4 @@ class Version0_3_0 extends Versioned {
   }
 }
 
-export default Version0_3_0;
+export default Version0_3_1;
```
Comment (rueshyna) [inline versioned/version0_3_1.ts:392]: Error:

example: `/KT1AxG32ZKBDazFXv2PxiL2HCJJm7MRs8cxg/history`

```
Unhandled Runtime Error
InvalidHexStringError: Invalid hex string "[object Object]" : Expecting even number of characters.

Source
versioned/version0_3_1.ts (316:33) @ content

  314 | status: "Non-executed;",
  315 | meta: content.execute_lambda.metadata
> 316 |   ? bytes2Char(content.execute_lambda.metadata)
      |               ^
  317 |   : "No meta supplied",
  318 | 
  319 | lambda: emitMicheline(
```
additional "Some" in the "content.execute_lambda.metadata"
for the version `< 0.3.1`, the "Some" seems to be handled by tzkt api.

File: versioned/version0_3_1.ts [hunk 2]
```diff
@@ -376,15 +377,18 @@ class Version0_3_1 extends Versioned {
...
     return {
       timestamp: prop.proposer.timestamp,
       author: prop.proposer.actor,
       status: status[Object.keys(prop.state)[0]!],
       content: prop.contents.map(this.mapContent),
-      signatures: [...Object.entries(prop.signatures)].map(([k, v]) => ({
-        signer: k,
-        result: v,
-      })),
+      signatures: [...prop.signatures.entries()].map(([k, v]) => {
```
Comment (rueshyna) [inline versioned/version0_3_1.ts:386]: bug:

`KT1EgKpn5HQmPwkbf4wcTShi6HHQPQFnGiEg/proposals`

```
Unhandled Runtime Error
TypeError: prop.signatures.entries is not a function or its return value is not iterable

Source
versioned/version0_3_1.ts (386:38) @ entries

  384 | status: status[Object.keys(prop.state)[0]!],
  385 | content: prop.contents.map(this.mapContent),
> 386 | signatures: [...prop.signatures.entries()].map(([k, v]) => {
      |                                ^
  387 |   return {
  388 |     signer: k,
  389 |     result: v,
```

File: pages/[walletAddress]/proposals.tsx [hunk 3]
```diff
@@ -113,9 +152,7 @@ const Proposals = () => {
       return;
     }
 
-    dispatch({ type: "setLoading", payload: true });
```
[the remark this answers is not in the document]
Comment (rueshyna) [reply pages/[walletAddress]/proposals.tsx:116]: It seems that the issue hasn't solved. Let's say I have an 1-of-2 wallet.
(1) I create a proposal on the new-proposal page.
(2) I go to proposal page to sign the proposal by click "sign" button.
(3) after signing, I'd like to see the proposal I signed show "resolve" button for me without refresh page.

[the remark this answers is not in the document]
Comment (JulesGuesnon) [reply pages/[walletAddress]/proposals.tsx:116]: I tried again, and it still works for me, so not sure why it's not working on your side 🤔 Could you send me the address of the contract + a video

**Review by rueshyna — APPROVED** [review]

**Pull request merged**